src/api/api.ts

```diff
@@ -1 +1,76 @@
+import { NewPostBody } from '../types/Post';
+
 export const BASE_URL = 'https://mate.academy/students-api';
+
+export async function getAllPosts() {
+  try {
+    const response = await fetch(`${BASE_URL}/posts`);
+    const data = await response.json();
+
+    return data;
+  } catch (error) {
+    return `Error: ${error}`;
+  }
+}
+
+export async function getUserPosts(userId: number) {
+  try {
+    // eslint-disable-next-line
+    const response = await fetch(`${BASE_URL}/posts${userId === 0 ? '' : '?userId=' + userId}`);
+    const data = await response.json();
+
+    if (!data) {
+      throw new Error('Dont find this user');
+    }
+
+    return data;
+  } catch (error) {
+    return `Error: ${error}`;
+  }
+}
+
+export async function getPostDetails(postId: number) {
+  try {
+    // eslint-disable-next-line
+    const response = await fetch(`${BASE_URL}/posts${postId && '/' + postId}`);
+    const data = await response.json();
+
+    if (!data) {
+      throw new Error('Dont find this user');
+    }
+
+    return data;
+  } catch (error) {
+    return `Error: ${error}`;
+  }
+}
+
+export async function getPostComments(postId: number) {
+  try {
+    // eslint-disable-next-line
+    const response = await fetch(`${BASE_URL}/comments${postId === 0 ? '' : '?postId=' + postId}`);
```

Comment:
call should getPostComments only if you have posted, so I think this ternary is useless here